src/PluginClient.php

```diff
@@ -62,9 +65,20 @@ public function __construct($client, array $plugins = [], array $options = [])
      */
     public function sendRequest(RequestInterface $request)
     {
-        $promise = $this->sendAsyncRequest($request);
+        if (!($this->client instanceof HttpClient)) {
+            return $this->sendAsyncRequest($request)->wait();
+        }
+
+        $client = $this->client;
+        $pluginChain = $this->createPluginChain($this->plugins, function (RequestInterface $request) use ($client) {
```

Comment:
Technically, this doesn't make any difference IMO. When a non-async client is passed, it is emulated to be an async client, which means you are free to always rely on sendAsyncRequest. If it is an async client, the upper condition would just return with an async client call. If it is not an async client, then the emulated async request send method will just do the same as this callback does.

What's the difference then?